platform/o.n.swing.plaf/src/org/netbeans/swing/plaf/windows8/DPISafeBorder.java

```diff
@@ -44,16 +44,26 @@
      * @param color may not be null
      */
     public static Border matte(int top, int left, int bottom, int right, Color color) {
-        return new DPISafeBorder(new Insets(top, left, bottom, right), color);
+        return fromDelegate(new MatteBorder(new Insets(top, left, bottom, right), color));
     }
 
-    private DPISafeBorder(Insets insets, Color color) {
-        if (insets == null)
+    public static Border fromDelegate(Border delegate) {
+        if (delegate == null)
```

Comment:
Should use curly braces(`{}`) although existing code doesn't use them.

Reply:
Well, my humble opinion is that we shall avoid throwing NullPointerExceptions directly from the code. It would be better to do an ```assert <whatever> != null;``` instead.

Reply:
Will add curly braces... I keep forgetting the code style. (The previous code was mine, too.)

Assertions can be disabled, which make them more or less useless for enforcing invariants--no way to know down the line whether they were enabled or not (e.g. when reading a log file while trying to debug an intermittent problem down the road).

Reply:
I agree - I tried to run netbeans with `substance` laf (or one of its siblings) and the author decided to enforce the "things interacting with GUI have to run on the EDT" rule. I ended badly (netbeans tries to do to much work off the EDT) and if Swing would have enforced the rule, netbeans could not have gone down the path in the first place. Now it is difficult to get rid of this.

The Code can be made better readable by using `java.util.Objects#requireNonNull`, then the missing braces will also not be a problem :-)

Reply:
Maybe, we can also use `@NonNull` to make "non-null" clear ? e.g. `public static Border fromDelegate(@NonNull Border delegate)`

Reply:
For the annotations we'd need a dependency on `Common Annotations`. That modules causes headaches: https://github.com/apache/netbeans/pull/1764